tests/tag_tests.rb

```diff
@@ -0,0 +1,49 @@
+class TagTest < MiniTest::Test
+
+  def test_create_a_new_tag
+    user = User.create(first_name: 'first', last_name: 'last', email: 'email@email.com')
+    tag = Tag.create
+    tag2 = Tag.create(name: 'tag', taggable: user)
+    refute tag.save
+    assert tag.errors.full_messages.include?("Name can't be blank")
+    assert tag.errors.full_messages.include?("Taggable can't be blank")
+    assert tag2.persisted?
+  end
+
+  def test_a_lesson_can_be_tagged
+    lesson = Lesson.create(name: 'lesson')
+    tag = lesson.tags.create(name: 'lesson_tag')
```

Comment:
Nice find! Creating/initializing through the proxy comes in _really_ handy later when we want to ensure we only find or create objects for, say, the `current_user`